core/field_angle.ts

```diff
@@ -546,4 +523,41 @@ Css.register(`
 
 fieldRegistry.register('field_angle', FieldAngle);
 
-(FieldAngle.prototype as AnyDuringMigration).DEFAULT_VALUE = 0;
\ No newline at end of file
+(FieldAngle.prototype as AnyDuringMigration).DEFAULT_VALUE = 0;
+
+/**
+ * The two main modes of the angle field.
+ * Compass specifies:
+ *   - clockwise: true
+ *   - offset: 90
+ *   - wrap: 0
+ *   - round: 15
+ *
+ * Protractor specifies:
+ *   - clockwise: false
+ *   - offset: 0
+ *   - wrap: 0
+ *   - round: 15
+ */
+export enum Mode {
```

Comment:
[nit] I tend to see and prefer to have interfaces/enums/types at the top of a file (defined before their use) instead of the bottom, although there's no functional problem with them being at the bottom. It's ultimately just a preference, though, so no big deal if you'd rather leave it as is.

Reply:
We've decided as a team not to have an opinion on the ordering of top-level declarations. (some people wanted types at the top, and other people wanted the "main" class/function/etc at the top). So I'm going to go ahead and merge as is.